common/src/main/java/io/netty/util/concurrent/MockTicker.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A fake {@link Ticker} that allows the caller control the flow of time.
+ * This can be useful when you test time-sensitive logic without waiting for too long
+ * or introducing flakiness due to non-deterministic nature of system clock.
+ */
+public interface MockTicker extends Ticker {
```

Comment:
imho it is kind of odd to see a `Mock*` outside of a test-jar.

Reply:
I don't really want to change the design or naming from the 5.0 version

Reply:
Just a general comment; We can make whatever changes we want and forward port back to 5.0.